hooks/gen_docs.py

```diff
@@ -1,13 +1,340 @@
-"""Generates the whole 'generatable' KPOps documentation"""
+"""Generates the whole 'generatable' KPOps documentation."""
+import csv
+import shutil
 import subprocess
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+from textwrap import fill
+from typing import Any, TypeVar
+
+from pytablewriter import MarkdownTableWriter
+from typer.models import ArgumentInfo, OptionInfo
 
 from hooks import PATH_ROOT
+from kpops.cli import main
+from kpops.cli.pipeline_config import PipelineConfig
 
 PATH_KPOPS_MAIN = PATH_ROOT / "kpops/cli/main.py"
 PATH_CLI_COMMANDS_DOC = PATH_ROOT / "docs/docs/user/references/cli-commands.md"
+PATH_DOCS_RESOURCES = PATH_ROOT / "docs/docs/resources"
+PATH_DOCS_VARIABLES = PATH_DOCS_RESOURCES / "variables"
+
+#####################
+# EXAMPLES          #
+#####################
+
+COMMENT_SYMBOL = "#"
+
+T = TypeVar("T", bound="EnvVar")
+@dataclass
+class EnvVar:
+    """Environment variable's properties."""
+
+    name: str
+    default_value: Any
+    required: bool
+    description: str | None
+    corresponding_setting_name: str | None
+
+    @classmethod
+    def from_record(cls: T, record: dict[str, Any]) -> T:
```

Comment:
clever use of the TypeVar :) alternatively you can use `typing.Self` for the return type (has to be imported from typing_extensions on Python 3.10)

Reply:
Done, much cleaner :)